apps/app/src/pages/Repository/Tests.tsx

```diff
@@ -0,0 +1,563 @@
+import { PageLoader } from "@/ui/PageLoader";
+import { Time } from "@/ui/Time";
+import { MagicTooltip } from "@/ui/Tooltip";
+
+import { NotFound } from "../NotFound";
+import {
+  SelectedTestsStateProvider,
+  useSelectedTestsState,
+} from "./SelectedTestsState";
+
+const RepositoryTestsQuery = graphql(`
+  query FlakyTests_repository_tests(
+    $ownerLogin: String!
+    $repositoryName: String!
+    $after: Int!
+    $first: Int!
+  ) {
+    repository(ownerLogin: $ownerLogin, repositoryName: $repositoryName) {
+      id
+      tests(first: $first, after: $after) {
+        pageInfo {
+          totalCount
+          hasNextPage
+        }
+        edges {
+          id
+          name
+          buildName
+          status
+          resolvedDate
+          mute
+          muteUntil
+          stabilityScore
+          lastSeen
+          unstable
+          dailyChanges {
+            date
+            count
+          }
+          totalBuilds
+          screenshot {
+            id
+            url
+            width
+            height
+          }
+        }
+      }
+    }
+  }
+`);
+
+const MuteTestsMutation = graphql(`
+  mutation muteTests($ids: [String!]!, $muted: Boolean!, $muteUntil: String) {
+    muteTests(ids: $ids, muted: $muted, muteUntil: $muteUntil) {
+      ids
+      mute
+      muteUntil
+    }
+  }
+`);
+
+type RepositoryTestsDocument = DocumentType<typeof RepositoryTestsQuery>;
+type Tests = NonNullable<RepositoryTestsDocument["repository"]>["tests"];
+type Test = Tests["edges"][0];
+type Screenshot = Tests["edges"][0]["screenshot"];
+
+const secondaryColumnClassNames: string[] = [
+  "w-40 hidden xl:flex",
+  "w-24 hidden lg:flex",
+  "w-32",
+];
+
+const Thumbnail = ({ screenshot }: { screenshot: Screenshot }) => {
+  const screenshotSrc = screenshot?.url
+    ? `${screenshot.url}?tr=w-${screenshot.width},h-${screenshot.height},c-at_max,dpr-2`
+    : null;
+  return (
+    <div className="h-16 w-28 min-w-fit overflow-hidden rounded border border-border">
+      {screenshotSrc && (
+        <img
+          src={screenshotSrc}
+          className="w-28"
+          style={{
+            aspectRatio: `${screenshot!.width}/${screenshot!.height}`,
+          }}
+        />
+      )}
+    </div>
+  );
+};
+
+const MuteIndicator = ({
+  mute,
+  muteUntil,
+}: {
+  mute: boolean;
+  muteUntil: string;
+}) => {
+  return mute ? (
+    <div className="flex h-5 items-center">
+      <MagicTooltip tooltip={`Muted until: ${moment(muteUntil).format("LLL")}`}>
+        <SpeakerXMarkIcon className="h-4 w-4 text-danger-500" />
+      </MagicTooltip>
+    </div>
+  ) : null;
+};
+
+const BuildNameField = ({ test }: { test: Test }) => {
+  return test.buildName !== "default" ? (
+    <div className="whitespace-nowrap">Build: {test.buildName}</div>
+  ) : null;
+};
+
+const LastSeenField = ({ test }: { test: Test }) => {
+  return (
+    <MagicTooltip tooltip={`Last seen: ${moment(test.lastSeen).format("LLL")}`}>
```

Comment:
`lastSeen` can be null, I fixed the generated types. It was considered as `any` that's why you didn't catch it.

<img width="309" alt="Screenshot 2023-03-23 at 21 15 35" src="https://user-images.githubusercontent.com/266302/227342394-3a92f96f-4541-4ca3-8816-fc02e7bc4672.png">

Ironical that the only wrong type results in a bug.